homeassistant/components/cookidoo/__init__.py

```diff
@@ -0,0 +1,70 @@
+"""The Cookidoo integration."""
+
+from __future__ import annotations
+
+from cookidoo_api import (
+    DEFAULT_COOKIDOO_CONFIG,
+    Cookidoo,
+    CookidooAuthException,
+    CookidooRequestException,
+    get_localization_options,
+)
+
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import CONF_LOCALIZATION, DOMAIN
+from .coordinator import CookidooConfigEntry, CookidooDataUpdateCoordinator
+from .helpers import cookidoo_localization_for_key
+
+PLATFORMS: list[Platform] = [Platform.TODO]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: CookidooConfigEntry) -> bool:
+    """Set up Cookidoo from a config entry."""
+
+    email = entry.data[CONF_EMAIL]
+    password = entry.data[CONF_PASSWORD]
+    localization = cookidoo_localization_for_key(
+        await get_localization_options(), entry.data[CONF_LOCALIZATION]
+    )
+    session = async_get_clientsession(hass)
+    cookidoo = Cookidoo(
+        session,
+        {
+            **DEFAULT_COOKIDOO_CONFIG,
+            "localization": localization,
+            "email": email,
+            "password": password,
+        },
+    )
+
+    try:
+        await cookidoo.login()
+    except CookidooRequestException as e:
+        raise ConfigEntryNotReady(
+            translation_domain=DOMAIN,
+            translation_key="setup_request_exception",
+        ) from e
+    except CookidooAuthException as e:
+        raise ConfigEntryAuthFailed(
```

Comment:
I feel like it would be cleaner if this was in `_async_setup` in the coordinator

Reply:
Do you think so @zweckj? I based it on other integrations such as `bring` which is also like this. No problem to change it, just wondering what the differences are.

Reply:
When putting it into the coordinator, I am missing the email for the exception. In the coordinator it is only available after the initial config init. This will make it more ugly in my opinion.

Reply:
Therefore I would rather leave it be as it is.

Reply:
the coordinator automatically has a `self.config_entry` (after super().init) so you should have everything there you need. Bring was probably written before that was a thing. You can probably cut paste the code in `_async_setup` and replace the entry with self.config_entry

Reply:
but it's a personal preference in the end. I just feel like then all the interactions with the API are in the coordinator, which is nice.

Reply:
Interesting, the IQS example though still does test before setup in the `__init__.py` https://developers.home-assistant.io/docs/core/integration-quality-scale/rules/test-before-setup 
Maybe someone (@joostlek ) should update the example?

Reply:
@tr4nt0r ot to this PR: but that example does not have a coordinator